src/OSPSuite.Core/Domain/Services/PKValuesCalculator.cs

```diff
@@ -133,52 +143,76 @@ private static PKCalculationOptions checkOptions(PKCalculationOptions options)
          return options ?? new PKCalculationOptions();
       }
 
-      public PKValues CalculatePK(DataColumn dataColumn, PKCalculationOptions options = null)
+      public PKValues CalculatePK(DataColumn dataColumn, PKCalculationOptions options = null, IReadOnlyList<DynamicPKParameter> dynamicPKParameters = null)
       {
-         return CalculatePK(dataColumn.BaseGrid.Values, dataColumn.Values, options);
+         return CalculatePK(dataColumn.BaseGrid.Values, dataColumn.Values, options, dynamicPKParameters);
       }
 
-      private IEnumerable<DosingInterval> allIntervalsFor(IReadOnlyList<float> time, IReadOnlyList<float> concentration, PKCalculationOptions options)
+      private void addDynamicPKValues(ICache<string, double> pk, IReadOnlyList<DynamicPKParameter> dynamicPKParameters, List<float> time, List<float> concentration, PKCalculationOptions options)
       {
-         var timeValues = time.ToList();
-         var concentrationValues = concentration.ToList();
-
-         if (!timeValues.Any() || !concentrationValues.Any())
-            yield break;
+         foreach (var dynamicPKParameter in dynamicPKParameters)
```

Comment:
Dyanmic parameters. One entry for each. The user is responsible to ensure unicity in names